src/coreclr/vm/callhelpers.cpp

```diff
@@ -471,7 +471,11 @@ void MethodDescCallSite::CallTargetWorker(const ARG_SLOT *pArguments, ARG_SLOT *
                     case 1:
                     case 2:
                     case 4:
+#if defined(TARGET_LOONGARCH64)
```

Comment:
If this is trying to extend the value to 64-bit, I do not think it is sufficient. The value should be sign or zero extended depending on whether the argument is signed or unsigned.

Reply:
> If this is trying to extend the value to 64-bit, I do not think it is sufficient. The value should be sign or zero extended depending on whether the argument is signed or unsigned.

Maybe this is a another question for different types CPU-architecture.
For AMD64 and ARM64 which they are 64-bits processors, the operation of the 32-bits values usually does not care the hight 32-bits.
But for the LoongArch64,MIPS64 (maybe the RISC-V-64),  the operation of the 32-bits values usually must care the hight 32-bits.

By the way, now the CoreCLR seems uncovered the non-AMD64/ARM64 type architectures.
Liking the comparation instruction, the operation's restriction for common data types.

Reply:
Yes, the ABIs for existing CoreCLR architectures has unspecified value for the padding bits when primitive types are passed in registers. LoongArch is the first architecture that requires predictable content of the padding bits. This will likely require fixes in number of places.

In places like this one that are not performance sensitive, it should be fine to initialize the padding bits unconditionally (without ifdef). It will make the code simpler.

Also, I expect that this will be a source of subtle bugs for DllImport signatures. It is not unusual to see DllImport signature with signed/unsigned mismatch in .NET code out there (e.g. .NET side is defined as signed `int` and unmanaged C side is defined as `unsigned int`). DllImport with signed/unsigned mismatch works fine on all existing CoreCLR architectures today, but it won't be the case for LoongArch. Did you happen to run into any DllImport bugs like this during the port?

Reply:
> Yes, the ABIs for existing CoreCLR architectures has unspecified value for the padding bits when primitive types are passed in registers. LoongArch is the first architecture that requires predictable content of the padding bits. This will likely require fixes in number of places.
> 
> In places like this one that are not performance sensitive, it should be fine to initialize the padding bits unconditionally (without ifdef). It will make the code simpler.

good idea.
Thanks.

> 
> Also, I expect that this will be a source of subtle bugs for DllImport signatures. It is not unusual to see DllImport signature with signed/unsigned mismatch in .NET code out there (e.g. .NET side is defined as signed `int` and unmanaged C side is defined as `unsigned int`). DllImport with signed/unsigned mismatch works fine on all existing CoreCLR architectures today, but it won't be the case for LoongArch. Did you happen to run into any DllImport bugs like this during the port?

Yes, you are right.
If you had write a project, e.g. implenmened by C/C++, while you just considered the x64/arm64 and didn't care about 64the sign/unsign int type,  this may be ok on x64/arm64. But maybe failded on LoongArch64/MIPS64/RISC-V-64.
We had tested the CoreCLR calling the so libraries on linux, if the C# and the C/C++ are same defination, the arguments are the same with sigin-int or unsign-int at the same time, that will be ok.